R/util.R

```diff
@@ -142,3 +142,35 @@ lookupByName = function(listpath, name, prop = "name") {
   assign(key, val, envir = byNameHash)
   val
 }
+
+#' Find a Resource By Name
+#'
+#' Allows createion of a by-experiment lookup object, which can
+#' then be used to find resources by name instead of id. Utilizes
+#' the internal lookupByName cache to improve performance.
+#' The resource's id is cached for the duration of the R
+#' session. Resources such as gates that exist within an experiment
+#' are cached within the experiment's scope.
+#'
+#' \code{
+#' lookup = createLookup("5e1f66a06f5f3f0759b479c9")
+#' lookup("gates", "test_gate")
+#'
+#' @param experimentId Experiment to create lookup for.
+#' Internal function params:
+#' @param resource Type of resource.
```

Comment:
I'd list the available options and/or say that they have to be pluralized.

Probably also add a validator for the `resource` argument -- `resource %in% c("gates", ...)`